static/app/views/performance/metricReadout.tsx

```diff
@@ -0,0 +1,109 @@
+import type {ReactText} from 'react';
+import styled from '@emotion/styled';
+
+import Duration from 'sentry/components/duration';
+import FileSize from 'sentry/components/fileSize';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
+import {Tooltip} from 'sentry/components/tooltip';
+import {defined} from 'sentry/utils';
+import type {CountUnit} from 'sentry/utils/discover/fields';
+import {DurationUnit, RateUnit, SizeUnit} from 'sentry/utils/discover/fields';
+import {formatAbbreviatedNumber, formatRate} from 'sentry/utils/formatters';
+import {Block} from 'sentry/views/starfish/views/spanSummaryPage/block';
+
+type Unit = DurationUnit.MILLISECOND | SizeUnit.BYTE | RateUnit | CountUnit;
+
+interface Props {
+  title: string;
+  unit: Unit;
+  value: ReactText | undefined;
+  align?: 'left' | 'right';
+  isLoading?: boolean;
+  tooltip?: React.ReactNode;
+}
+
+export function MetricReadout({
+  unit,
+  value,
+  title,
+  tooltip,
+  align = 'right',
+  isLoading,
+}: Props) {
+  return (
+    <Block title={title} alignment={align}>
+      {(() => {
```

Comment:
l: My personal opinion is that the IIFE makes this less readable. I'd would prefer if this whole anonymous function closure was moved into it's own component.

Reply:
Yeah, IIFEs are pretty divisive. I don't have a strong opinion here (I guess I expressed a weak opinion when I wrote it as an IIFE) so I changed it to a component instead 👍🏻